sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/typedaggregators.scala

```diff
@@ -99,3 +97,67 @@ class TypedAverage[IN](val f: IN => Double) extends Aggregator[IN, (Double, Long
     toColumn.asInstanceOf[TypedColumn[IN, java.lang.Double]]
   }
 }
+
+class TypedMinDouble[IN](val f: IN => Double) extends Aggregator[IN, Double, Double] {
+  override def zero: Double = Double.PositiveInfinity
```

Comment:
@viirya I just had a go at your suggestion, but it seems to be more complicated than anticipated. Spark performs some implicit casts (I think as part of Catalyst) between `java.lang.Double` and `scala.Double`, causing a nullpointer:
`java.lang.NullPointerException at scala.Predef$.Double2double(Predef.scala:365!`
I am not sure if this method is feasible. 

Sample of the `merge` function:
`override def merge(b1: java.lang.Double, b2: java.lang.Double): java.lang.Double =  java.lang.Math.min(b1, b2)`